File: src/gestorviajes/Transportadora.java
package gestorviajes;
import java.io.Serializable;
import java.util.ArrayList;

import gestortaller.*;
public class Transportadora implements Serializable{

	private static final long serialVersionUID = 1L;
	
	private String nombre;
	private ArrayList<Terrestre> terrestres = new ArrayList<Terrestre>();
	private ArrayList<Viaje> viajes = new ArrayList<Viaje>();
	private static ArrayList<Transportadora> transportadoras = new ArrayList<Transportadora>();

	//CONSTRUCTOR
	public Transportadora(String nombre) {
		this.nombre = nombre;
		transportadoras.add(this);
	}

	@Override
	public String toString() {
		return this.nombre;
	}


	public static Transportadora buscarTransportadoraPorNombre(String nombre2)
	{
		Transportadora retorno = null;
		for (int i = 0; i <Transportadora.getTransportadoras().size(); i++)
		{
			if( Transportadora.getTransportadoras().get(i).getNombre().equalsIgnoreCase(nombre2))
			{
				retorno=  Transportadora.getTransportadoras().get(i);
			}
		}
		return retorno;
	}


	public Viaje buscarViajePorID (ArrayList<Viaje> viajes, int ID)
	{
		for (int i = 0; i < viajes.size(); i++)
		{
		  if (viajes.get(i).getID() == ID )
		  {
			  return viajes.get(i);
		  }
		}
		return null;
	}
	
	
	public Viaje buscarViajePorTerrestre (ArrayList<Viaje> viajes, String nombre_Terrestre)
	{
		for (int i = 0; i < viajes.size(); i++)
		{
		  if (viajes.get(i).getTerrestre().getNombre().equals(nombre_Terrestre) )
		  {
			  return viajes.get(i);
		  }
		}
		return null;
	}
	
	
	public ArrayList<Viaje> buscarViajePorDestino (ArrayList<Viaje> viajes, String destino)
	{
		ArrayList<Viaje> viajesPorDestino = new ArrayList<Viaje>();
		for (int i = 0; i < viajes.size(); i++)
		{
		  if (viajes.get(i).getDestino().equalsIgnoreCase(destino)) 
		  {
			  viajesPorDestino.add(viajes.get(i));
		  }
		}
		return viajesPorDestino;
	}

	public ArrayList<Viaje> buscarViajePorFecha (ArrayList<Viaje> viajes, String fecha)
	{
		ArrayList<Viaje> viajesPorFecha = new ArrayList<Viaje>();
		for (int i = 0; i < viajes.size(); i++)
		{
		  if (viajes.get(i).getFecha_de_salida().equals(fecha))
		  {
			  viajesPorFecha.add(viajes.get(i));
		  }
		}
		return viajesPorFecha;
	}

	

	public ArrayList<Viaje> viajesDisponibles(ArrayList<Viaje> viajes)
	{
		ArrayList<Viaje> viajesDisponibles = new ArrayList<Viaje>();
		for (int i = 0; i < viajes.size(); i++)
		{
			if (!viajes.get(i).isEstaCompleto())
			{
				viajesDisponibles.add(viajes.get(i));
			}
		}
		return viajesDisponibles;
	}

	public void agregarViaje(Viaje viaje)
	{
		viajes.add(viaje);
	}

	
	public Boolean cancelarViaje(int viaje_a_eliminar)
	{
		for (int i = 0; i < viajes.size(); i++)
		{
		  if (viajes.get(i).getID() == viaje_a_eliminar )
		  {
			  viajes.remove(i);
			  return true;
		  }
		}
		return false;
	}

	
	public static Tiquete BuscarTiquete(int ID)
	{
		ArrayList<Transportadora> transportadorasDisponibles = Transportadora.getTransportadoras();
		for (int i = 0; i < transportadorasDisponibles.size(); i++)
		{
			Transportadora transportadora = transportadorasDisponibles.get(i);
			for (int j = 0; j < transportadora.getViajes().size(); j++)
			{

				Viaje viaje = transportadora.getViajes().get(j);
				Tiquete tiquete_buscado = viaje.buscarTiquetePorID(viaje.getTiquetes(), ID);
				if (tiquete_buscado != null)
				{
					return tiquete_buscado;
				}
			}
		}
		return null;
	}
	  

	public String getNombre() {
		return nombre;
	}


	public void setNombre(String nombre) {
		this.nombre = nombre;
	}


	public ArrayList<Viaje> getViajes() {
		return viajes;
	}


	public void setViajes(ArrayList<Viaje> viajes) {
		this.viajes = viajes;
	}

	public ArrayList<Terrestre> getTerrestres() {
		return terrestres;
	}

	public void setBuses(ArrayList<Terrestre> buses) {
		this.terrestres = buses;
	}

	public static ArrayList<Transportadora> getTransportadoras() {
		return transportadoras;
	}

	public static void setTransportadoras(ArrayList<Transportadora> transportadoras) {
		Transportadora.transportadoras = transportadoras;
	}
}

File: practica02/Practica2VALE/Practica02VALE/SRC/gestorAplicacion/adminViajes/Transportadora.py
from excepciones.ErrorAsignacion import ExcepcionIdTiquete
from gestorAplicacion.parada import *
from gestorAplicacion.adminViajes.Viaje import Viaje
from gestorAplicacion.adminViajes.Tiquete import Tiquete

#se guarda la inf de todas las transportadoras
class Transportadora():

    #ATRIBUTOS
    _transportadoras = []

    #CONSTRUCTOR
    def __init__(self, nombre):
        self._terrestres = []
        self._viajes = []

        self._nombre = nombre
        Transportadora._transportadoras.append(self)

    def toString(self):
        return self._nombre



#buscar Transportadora por nombre

    @staticmethod
    def buscarTransportadoraPorNombre(nombre2):
        retorno = None
        i = 0
        while i < len(Transportadora.getTransportadoras()):
            if Transportadora.getTransportadoras()[i].getNombre().lower() == nombre2.lower():
               
                retorno = Transportadora.getTransportadoras()[i]
            i += 1
        return retorno

#buscar viaje por id 
    def buscarViajePorID(self, viajes, ID):
        i = 0
        while i < len(viajes):
            if viajes[i].getID() == ID:
                return viajes[i]
            i += 1
        return None

#buscar viaje por terrestre 
    def buscarVueloPorTerrestre(self, vuelos, nombre_Terrestre):
        i = 0
        while i <len(viajes):
            if viajes [i].getTerrestre().getNombre().casefold() ==nombre_Terrestre.casefold():
                return viajes[i]
            i += 1
        return None

    
    def buscarViajePorDestino(self, viajes, destino):
        viajesPorDestino = []
        i = 0
        while i < len(viajes):
            if viajes[i].getDestino().casefold() == destino.casefold():
                viajesPorDestino.append(viajes[i])
            i += 1
        return viajesPorDestino

    
#buscar viaje por fechas 
    def buscarViajePorFecha(self, viajes, fecha):
        viajesPorFecha = []
        i = 0
        while i < len(viajes):
            if viajes[i].getFecha_de_salida() == fecha:
                viajesPorFecha.append(viajes[i])
            i += 1
        return viajesPorFecha

   
#viajes disponibles 
    def viajesDisponibles(self, viajes):
        viajesDisponibles = []
        i = 0
        while i < len(viajes):
            if not viajes[i].isEstaCompleto():
                viajesDisponibles.append(viajes[i])
            i += 1
        return viajesDisponibles

    

#agregarviajes
    def agregarViaje(self, viaje):
        self._viajes.append(viaje)

#cancelar viaje recibe un entero
    def cancelarViaje(self, viaje_a_eliminar):
        i = 0
        while i < len(self._viajes):
            if self._viajes[i].getID() == viaje_a_eliminar:
                self._viajes.pop(i)
                return True
            i += 1
        return False
#buscar tiquete por id 


    @staticmethod
    def BuscarTiquete(ID):
        tiquete_buscado = None
        transportadorasDisponibles = Transportadora.getTransportadoras()
        i = 0
        while i < len(transportadorasDisponibles):
            transportadora = transportadorasDisponibles[i]
            j = 0
            while j < len(transportadora.getViajes()):

                viaje = transportadora.getViajes()[j]
                tiquete_buscado = viaje.buscarTiquetePorID(viaje.getTiquetes(), ID)
                if tiquete_buscado is not None:
                    return tiquete_buscado
                j += 1
            i += 1
        
        if tiquete_buscado == None:
            raise ExcepcionIdTiquete(ID)
        return None

   

    def getNombre(self):
        return self._nombre


    def setNombre(self, nombre):
        self._nombre = nombre


    def getViajes(self):
        return self._viajes


    def setViajes(self, viajes):
        self._viajes = viajes

    def getTerrestres(self):
        return self._terrestres

    def setBuses(self, buses):
        self._terrestres = buses

    @staticmethod
    def getAerolineas():
        return Transportadora._transportadoras

    @staticmethod
    def setTransportadoras(transportadoras):
        Transportadora._transportadoras = transportadoras
